vault/seal/awskms/awskms.go

```diff
@@ -98,6 +99,21 @@ func (k *AWSKMSSeal) SetConfig(config map[string]string) (map[string]string, err
 		k.region = region
 	default:
```

Comment:
Could we just remove this `default:` case? It seems to me that if the user does not specify the region in the config, and there is no ENV vars that cover it, then we'd be in the case of "detect this for me". Instead we set it a default of `us-east-1` for no clear reason to me. 

Omitting this default should allow the [credsConfig.GenerateCredentialChain SDK call in `getAWSKMSClient`](https://github.com/hashicorp/vault/blob/master/vault/seal/awskms/awskms.go#L280) to do it's magic and set the region automatically from the instance. By providing a default, we prevent that:

https://github.com/hashicorp/vault/blob/ea8dacf583b0a50cc347e07669345f5739b60292/helper/awsutil/generate_credentials.go#L66-L70

Reply:
OK, I see my error, I misread and got confused on where the GenerateCredentialChain was checking the ENV vars, in hindsight very silly 😄

Reply:
No worries, I took the exact same thought path! Thanks for looking at this PR!